mozci/data/sources/treeherder/__init__.py

```diff
@@ -35,6 +59,7 @@ def session(self):
         session.headers = {"User-Agent": "mozci"}
         return session
 
+    @lru_cache(maxsize=50)
```

Comment:
Why the cache here?

Reply:
I realized that if Treeherder is missing task ids (which seems to be the case sometimes), then we'll re-run the query again for each task id that is missing. This is a guard to ensure we aren't actually hitting Treeherder each time.